File: COMPONENTS/api/pkg/dtos/registerRequestDto.go
package dtos

// Data transfer object for a register request.
type RegisterRequestDto struct {
	Email    string `json:"email" validate:"required,email" example:"johndoe@example.com"`
	Username string `json:"username" validate:"required,min=3,max=32" example:"johndoe" minLength:"3" maxLength:"32"`
	Password string `json:"password" validate:"required,min=8,max=32" example:"password123" minLength:"8" maxLength:"32"`
}

File: COMPONENTS/app/app/src/main/java/com/dmarts05/speedshield/data/model/RegisterRequestDto.kt
package com.dmarts05.speedshield.data.model

import com.google.gson.annotations.SerializedName

data class RegisterRequestDto(
    @SerializedName("email")
    val email: String,
    @SerializedName("username")
    val username: String,
    @SerializedName("password")
    val password: String,
)
